test/test_ops.py

```diff
@@ -614,13 +616,15 @@ def _case_four_transform(t):
         #   tensor is a floating point or complex dtype.
         _dtypes = floating_and_complex_types_and(torch.float16, torch.bfloat16)
         if (isinstance(expected, torch.Tensor) and expected.dtype in _dtypes or
-                (not isinstance(expected, torch.Tensor) and
-                 reduce(lambda cur, t: cur or t.dtype in _dtypes, expected, False))):
+                (not isinstance(expected, torch.Tensor) and any(t.dtype in _dtypes for t in expected))):
             def _case_five_transform(t):
                 return make_tensor(t.shape, dtype=torch.long, device=t.device)
 
-            out = out = _apply_out_transform(_case_five_transform, expected)
-            with self.assertRaises(RuntimeError):
+            out = _apply_out_transform(_case_five_transform, expected)
+            msg_fail = "" if not isinstance(expected, torch.Tensor) else \
+                       ("Expected RuntimeError when doing an unsafe cast from an out "
```

Comment:
This message is a little misleading because it's when doing an unsafe cast from a computation that produces a floating point or complex value to an out tensor with long dtype.

So it's not a cast FROM torch.long. It's a cast from a floating point or complex dtype to torch.long, which should be prohibited (because that cast is unsafe).

Reply:
Ooops. Corrected now.